main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -135,6 +135,10 @@ object Unification {
 
     case (_, _: Type.AssocType) => Result.Ok(Substitution.empty, List(Ast.BroadEqualityConstraint(tpe1, tpe2)))
 
+    case (Type.Cst(TypeConstructor.Error(k1), _), t2) if k1 == t2.kind => Result.Ok(Substitution.empty, Nil)
```

Comment:
This might be too strict. `KindUnification` is probably better here.

Reply:
Why? Now we infer the kinds.

Reply:
Because kind inference is not complete. I believe there are a few cases where Kind.Wild gets by.

Reply:
Hmm